lightningd/channel_control.c

```diff
@@ -386,21 +388,18 @@ static void sendfunding_done(struct bitcoind *bitcoind UNUSED,
 	tal_free(cs);
 }
 
-static void handle_funding_tx(struct channel *channel,
-			      const u8 *msg)
+static void send_funding_tx(struct channel *channel,
+			    const struct wally_tx *wtx TAKES)
 {
-	struct channel_send *cs;
 	struct lightningd *ld = channel->peer->ld;
+	struct channel_send *cs;
 
 	cs = tal(channel, struct channel_send);
 	cs->channel = channel;
-
-	if (!fromwire_channeld_funding_tx(tmpctx, msg, &cs->wtx)) {
-		channel_internal_error(channel,
-				       "bad channeld_funding_tx: %s",
-				       tal_hex(tmpctx, msg));
-		return;
-	}
+	if (taken(wtx))
+		cs->wtx = tal_steal(cs, wtx);
+	else
+		cs->wtx = wtx;
```

Comment:
If we're not supposed to take it, we should almost certainly clone it?

Reply:
Good point!